vendor/github.com/contiv/libOpenflow/util/stream.go

```diff
@@ -35,6 +37,8 @@ type MessageStream struct {
 	pool *BufferPool
 	// Message parser
 	parser Parser
+	// Channel to shut down the parser goroutine
+	parserShutdown chan bool
```

Comment:
Looking at the original PR/discussion might give more context:  https://github.com/contiv/libOpenflow/pull/1

The tests are also in that PR